### DISCUSSION TITLE
Thank you.

### DISCUSSION THREAD

**Discussion by aimgdev** [hydrostack/hydro#13] [General]
Thank you for doing this project.
Even though it's really new I'm going to try to port my production web app to this framework because I believe it provides a superior user experience vs Blazor while retaining many of the development benefits. Hoping to help and become a sponsor soon.

What is the behavior if a front-end user switches the backend server? For example during an upgrade or one of the servers is down and the load balancer switches.

**Comment by kjeske** [comment]
Hello! Thank you for kind words and for giving Hydro a try.

The load balancing will work fine, switching between the nodes won't cause issues.

The only thing that is required in general by ASP.NET Core to work in load balanced environment is to keep the protection keys in a common place. Protection keys are used for encrypting authentication cookies, anti-forgery tokens, and also Hydro utilizes some of its features. [Here more details](https://usehydro.dev/advanced/load-balancing.html). Are you already taking care of the protection keys?
